src/ion/data_structures.rs

```diff
@@ -432,6 +447,17 @@ pub struct Env<'a, F: Function> {
     pub annotations_enabled: bool,
 }
 
+impl<'a, F: Function> Drop for Env<'a, F> {
+    fn drop(&mut self) {
```

Comment:
I don't think we actually need this, given that we know our key and index types are `Copy` (`u32`s in fact)?

Reply:
(Perhaps in trivial cases the empty drop impl means things optimize away, but I'd be a little skeptical of LLVM's ability to do that here in concert with the use of the drain iters etc)